sql/slave.cc

```diff
@@ -6184,25 +6184,6 @@ static int process_io_rotate(Master_info *mi, Rotate_log_event *rev)
     mi->events_till_disconnect++;
 #endif
 
-  /*
-    If description_event_for_queue is format <4, there is conversion in the
-    relay log to the slave's format (4). And Rotate can mean upgrade or
-    nothing. If upgrade, it's to 5.0 or newer, so we will get a Format_desc, so
-    no need to reset description_event_for_queue now. And if it's nothing (same
-    master version as before), no need (still using the slave's format).
-  */
-  if (mi->rli.relay_log.description_event_for_queue->binlog_version >= 4)
```

Comment:
Either way for me, but I would rather go conservatively  for a more "delicate"
`DBUG_ASSERT(mi->rli.relay_log.description_event_for_queue->binlog_version >= 4)` .

Reply:
Sounds good to me